### PR TITLE
CosmosClient.ReadAccountAsync: Fixes it to throw CosmosException instead of DocumentClientException

### PR DESCRIPTION
**Pull request by SchintaMicrosoft**
# Pull Request Template

## Description

I have tested the Readaccountasync method with invalid key. it is throwing document client exception instead of Cosmos Exception. So I try to catch that error in EnsureValidClientAsync method to convert it to Cosmos exception and I have added test method EnsureUnauthorized_ThrowsCosmosClientException_ReadAccountAsync in CosmosPermissionTests.cs .

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #2792

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs [hunk 1]
```diff
@@ -372,6 +372,26 @@ public async Task EnsureUnauthorized_ThrowsCosmosClientException()
...
+            {
+                CosmosException exception1 = await Assert.ThrowsExceptionAsync<CosmosException>(() => cosmosClient.ReadAccountAsync());
+                Assert.AreEqual(HttpStatusCode.Unauthorized, exception1.StatusCode);
+            }
+            catch (Exception ex)
+            {
+
+                string exx = ex.Message;
+            }
+
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs:393]: nit remove extra line

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs [hunk 2]
```diff
@@ -372,6 +372,26 @@ public async Task EnsureUnauthorized_ThrowsCosmosClientException()
...
+             string endpoint = ConfigurationManager.AppSettings["GatewayEndpoint"];
+            authKey = authKey.Replace("m", "M");
+            CosmosClient  cosmosClient = new CosmosClient(endpoint, authKey);
+
+            try
+            {
+                CosmosException exception1 = await Assert.ThrowsExceptionAsync<CosmosException>(() => cosmosClient.ReadAccountAsync());
+                Assert.AreEqual(HttpStatusCode.Unauthorized, exception1.StatusCode);
+            }
+            catch (Exception ex)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs:388]: Why is the try/catch needed?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs [hunk 3]
```diff
@@ -372,6 +372,26 @@ public async Task EnsureUnauthorized_ThrowsCosmosClientException()
             CosmosException exception = await Assert.ThrowsExceptionAsync<CosmosException>(() => cosmosClient.GetContainer("test", "test").ReadItemAsync<dynamic>("test", new PartitionKey("test")));
             Assert.AreEqual(HttpStatusCode.Unauthorized, exception.StatusCode);
         }
+        [TestMethod]
+        public async Task EnsureUnauthorized_ThrowsCosmosClientException_ReadAccountAsync()
+        {
+           string authKey = ConfigurationManager.AppSettings["MasterKey"];
+             string endpoint = ConfigurationManager.AppSettings["GatewayEndpoint"];
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs:379]: nit: fix indents

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs [hunk 4]
```diff
@@ -372,6 +372,26 @@ public async Task EnsureUnauthorized_ThrowsCosmosClientException()
             CosmosException exception = await Assert.ThrowsExceptionAsync<CosmosException>(() => cosmosClient.GetContainer("test", "test").ReadItemAsync<dynamic>("test", new PartitionKey("test")));
             Assert.AreEqual(HttpStatusCode.Unauthorized, exception.StatusCode);
         }
+        [TestMethod]
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosPermissionTests.cs:375]: nit: extra new line before `[TestMethod]`

File: Microsoft.Azure.Cosmos/src/DocumentClient.cs [hunk 5]
```diff
@@ -1442,6 +1442,12 @@ internal virtual async Task EnsureValidClientAsync(ITrace trace)
                     this.isSuccessfullyInitialized = true;
                     return;
                 }
+                catch (DocumentClientException ex)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/DocumentClient.cs:1445]: The try/catch should cover both await operations. Line 1468 could also throw a DocumentClientException.

File: Microsoft.Azure.Cosmos/src/DocumentClient.cs [hunk 6]
```diff
@@ -1459,8 +1465,22 @@ internal virtual async Task EnsureValidClientAsync(ITrace trace)
...
+                    await initTask;
+                    this.isSuccessfullyInitialized = true;
+                }
+                catch (DocumentClientException ex)
+                {
+                    throw Resource.CosmosExceptions.CosmosExceptionFactory.Create(
+                         dce: ex,
+                         trace: trace);
+                }
+                catch (Exception e)
```
[the remark this answers is not in the document]
Comment (SchintaMicrosoft) [reply Microsoft.Azure.Cosmos/src/DocumentClient.cs]: Yes Removed the Catch block to resolve the Tracing Issue.

**Review by j82w — APPROVED** [review]

**Review by ealsur — APPROVED** [review]

**Pull request merged**